libkbfs/bserver_remote.go

```diff
@@ -405,25 +380,39 @@ func (b *BlockServerRemote) batchDowngradeReferences(ctx context.Context,
 			}
 			nonces[BlockRefNonce(ref.Ref.Nonce)] = ref.LiveCount
 		}
-
-		//figure out the not-yet-deleted references
-		oldNotDoneLen := len(notDone)
+		// update the list of references to downgrade
 		notDone = b.getNotDone(contexts, doneRefs)
-		if !madeProgress {
-			madeProgress = len(notDone) < oldNotDoneLen
-		}
-		if !madeProgress {
-			return doneRefs,
-				errors.New("Made no progress after a batch downgrade call")
-		}
 
 		//if context is cancelled, return immediately
 		select {
 		case <-ctx.Done():
-			return doneRefs, ctx.Err()
+			finalError = ctx.Err()
+			return nil
 		default:
 		}
 
+		// check whether to backoff and retry
+		if err != nil {
+			// if error is of type throttle, retry
+			if _, ok := err.(BServerErrorThrottle); ok {
+				return err
+			}
+			// non-throttle error, do not retry here
+			finalError = err
+		}
+		return nil
+	}, backoff.NewExponentialBackOff())
+
+	// if backoff has given up retrying, return error
+	if throttleErr != nil {
+		return doneRefs, throttleErr
```

Comment:
This is not a change in behavior but I notice that the interface function comments for `RemoveBlockReference` and `ArchiveBlockReferences` don't mention that `liveCounts` can be non-nil when there's a fatal throttle error.  Can you please add that to the comments (and in `BlockOps` too) so callers will know they are getting some information back about previously-completed call?